src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -108,4 +114,45 @@ public String ingestDataset(String id, IngestRequestModel ingestRequestModel, Au
             .addParameter(JobMapKeys.DATASET_ID.getKeyName(), id)
             .submit();
     }
+
+    public String addDatasetAssetSpecifications(
+        String datasetId, AssetModel assetModel, AuthenticatedUserRequest userReq
+    ) {
+        String description = "Add dataset asset specification " + assetModel.getName();
+        Dataset dataset = retrieve(UUID.fromString(datasetId));
+
+        List<DatasetTable> datasetTables = dataset.getTables();
+        Map<String, DatasetTable> tablesMap = new HashMap<>();
+        datasetTables.forEach(table -> tablesMap.put(table.getName(), table));
+
+        List<DatasetRelationship> datasetRelationships = dataset.getRelationships();
+        Map<String, DatasetRelationship> relationshipMap = new HashMap<>();
+        datasetRelationships.forEach(relationship -> relationshipMap
+            .put(relationship.getId().toString(), relationship));
+
+       AssetSpecification assetSpecification = DatasetJsonConversion
```

Comment:
This is fine for now. I made a tech debt ticket (https://broadworkbench.atlassian.net/browse/DR-682) for cleaning up the use of Map in the Dataset object.

The ticket says: The hash map thing strikes me as a premature optimization. Yet, we will have schemas with hundreds of tables and looping through them to link things up is expensive. My suggestion is: instead of building maps, we should just create the objects as maps to begin with. Build the map as we pull the data from the database or from the input model. We do lookups in many places., but only in a few do we build a hash table for them.